Variables in testing directory are not automatically loaded on Windows

### Terraform Version

```shell
Terraform v1.7.3
on windows_amd64
```


### Terraform Configuration Files


```terraform
# main.tf
variable "required" {
  type     = string
  nullable = false
}

resource "terraform_data" "name" {
  input = var.required
}
```

```terraform
# tests/terraform.tfvars
required = "test"
```

```terraform
# tests/test.tftest.hcl
run "test" {
  assert {
    condition     = terraform_data.name.input != null
    error_message = "true"
  }
}
```


### Debug Output

[INFO]  Terraform version: 1.7.3
[DEBUG] using github.com/hashicorp/go-tfe v1.41.0
[DEBUG] using github.com/hashicorp/hcl/v2 v2.19.1
[DEBUG] using github.com/hashicorp/terraform-svchost v0.1.1
[DEBUG] using github.com/zclconf/go-cty v1.14.1
[INFO]  Go runtime version: go1.21.5
[INFO]  CLI args: []string{"<PATH>/terraform.exe", "test", "-test-directory=tests"}
[DEBUG] Attempting to open CLI config file: C:\Users\<USER>\AppData\Roaming\terraform.rc
[DEBUG] File doesn't exist, but doesn't need to. Ignoring.
[INFO]  Loading CLI configuration from C:\Users\<USER>\AppData\Roaming\terraform.d\credentials.tfrc.json
[DEBUG] ignoring non-existing provider search directory terraform.d/plugins
[DEBUG] ignoring non-existing provider search directory C:\Users\<USER>\AppData\Roaming\terraform.d\plugins
[DEBUG] ignoring non-existing provider search directory C:\Users\<USER>\AppData\Roaming\HashiCorp\Terraform\plugins
[INFO]  CLI command args: []string{"test", "-test-directory=tests"}
[DEBUG] checking for provisioner in "."
[DEBUG] TestSuiteRunner: found 1 files with 1 run blocks
[DEBUG] TestFileRunner: starting validate for tests\test.tftest.hcl/test
[DEBUG] Building and walking validate graph
[DEBUG] adding implicit provider configuration provider["terraform.io/builtin/terraform"], implied first by terraform_data.name
[DEBUG] ProviderTransformer: "terraform_data.name" (*terraform.NodeValidatableResource) needs provider["terraform.io/builtin/terraform"]
[ERROR] AttachSchemaTransformer: No provider config schema available for provider["terraform.io/builtin/terraform"]
[DEBUG] ReferenceTransformer: "terraform_data.name" references: [var.required]
[DEBUG] ReferenceTransformer: "var.required" references: []
[DEBUG] ReferenceTransformer: "provider[\"terraform.io/builtin/terraform\"]" references: []
[DEBUG] Starting graph walk: walkValidate
[DEBUG] TestFileRunner: completed validate for  tests\test.tftest.hcl/test


### Expected Behavior

Test to be successful

### Actual Behavior

Error: The module under test for run block "test" has a required variable "required" with no set value. Use a -var or -var-file command line argument or add this variable into a "variables" block within the test file or run block.

### Steps to Reproduce

`terraform test` or `terraform test -test-directory=tests`

### Additional Context

_No response_

### References

_No response_

--- a/internal/backend/local/test.go
+++ b/internal/backend/local/test.go
@@ -7,7 +7,7 @@ import (
 	"context"
 	"fmt"
 	"log"
-	"path"
+	"path/filepath"
 	"sort"
 	"time"
 
@@ -444,7 +444,7 @@ func (runner *TestFileRunner) run(run *moduletest.Run, file *moduletest.File, st
 				diags = diags.Append(tfdiags.Sourceless(
 					tfdiags.Warning,
 					"Failed to print verbose output",
-					fmt.Sprintf("Terraform failed to print the verbose output for %s, other diagnostics will contain more details as to why.", path.Join(file.Name, run.Name))))
+					fmt.Sprintf("Terraform failed to print the verbose output for %s, other diagnostics will contain more details as to why.", filepath.Join(file.Name, run.Name))))
 			} else {
 				run.Verbose = &moduletest.Verbose{
 					Plan:         plan,
@@ -532,7 +532,7 @@ func (runner *TestFileRunner) run(run *moduletest.Run, file *moduletest.File, st
 			diags = diags.Append(tfdiags.Sourceless(
 				tfdiags.Warning,
 				"Failed to print verbose output",
-				fmt.Sprintf("Terraform failed to print the verbose output for %s, other diagnostics will contain more details as to why.", path.Join(file.Name, run.Name))))
+				fmt.Sprintf("Terraform failed to print the verbose output for %s, other diagnostics will contain more details as to why.", filepath.Join(file.Name, run.Name))))
 		} else {
 			run.Verbose = &moduletest.Verbose{
 				Plan:         plan,
@@ -1252,7 +1252,7 @@ func (runner *TestFileRunner) initVariables(file *moduletest.File) {
 	for name, value := range runner.Suite.GlobalVariables {
 		runner.globalVariables[name] = value
 	}
-	if path.Dir(file.Name) == runner.Suite.TestingDirectory {
+	if filepath.Dir(file.Name) == runner.Suite.TestingDirectory {
 		// If the file is in the testing directory, then also include any
 		// variables that are defined within the default variable file also in
 		// the test directory.

